eslint.config.cjs

```diff
@@ -1,50 +1,194 @@
+    ],
     rules: {
+      "@typescript-eslint/no-explicit-any": "error", // Disallow 'any'
       "@typescript-eslint/no-unused-vars": "off",
-      "semi": ["warn", "never"],
-      "unicorn/filename-case": "off",
-      "unicorn/prefer-node-protocol": "off",
-      "unicorn/numeric-separators-style": "off",
-      "unicorn/no-hex-escape": "off",
-      "unicorn/better-regex": "off",
-      "unicorn/no-zero-fractions": "off",
-      "unicorn/no-array-for-each": "off",
-      "unicorn/explicit-length-check": "off",
-      "unicorn/no-process-exit": "off",
-      "unicorn/prefer-json-parse-buffer": "off",
-      "no-process-exit": "off",
-      "no-await-in-loop": "off",
-      "no-control-regex": "off",
+      "@typescript-eslint/typedef": "error",
+      
+      // stylistic rules
+      "@stylistic/array-bracket-spacing": ["error", "never"],
+      "@stylistic/array-bracket-newline": ["warn", { "multiline": true }],
+      "@stylistic/indent": ["warn", 2, { "SwitchCase": 1 }],
+      "@stylistic/max-statements-per-line": ["warn", { "max": 1, "ignoredNodes": ['BreakStatement'] }],
+      "@stylistic/no-multi-spaces": "warn", // Disallow multiple spaces except for alignment
+      "@stylistic/no-trailing-spaces": "warn",
+      "@stylistic/object-curly-spacing": ["warn", "never"],
+      "@stylistic/quotes": ["error", "single", { "avoidEscape": true }], // Allows double quotes (") when escaping single quotes
+      "@stylistic/semi": ["warn", "never"],
+
+      // eslint rules
+      "camelcase": "off",
+      "complexity": ["warn", 30],
+      "max-nested-callbacks": "warn",
+
+      "no-control-regex": "warn",
+      "no-console": "off",
+      "no-constant-condition": "warn",
       "no-undef": "off",
       "no-unused-vars": "off",
-      "no-console": "off",
-      "no-unused-expressions": "off",     
-      "no-constant-condition": "off",       
-      "max-nested-callbacks": "off",
-      "camelcase": "off",
-      "node/no-missing-import": "off",
-      "complexity": "off",
-      "indent": ["error", 2, { "SwitchCase": 1 }],
+      "no-unused-expressions": "error",
+      "no-await-in-loop": "off",
+
+      // eslint-plugin-n rules (these were moved from eslint-plugin-node)
+      "n/exports-style": [
+        "error",
+        "exports",
+        {
+          "allowBatchAssign": false
+        }
+      ],
+      "n/no-missing-import": "off",
+      "n/no-process-exit": "off",
+      "n/no-unpublished-import": ["error", {
+        "ignoreTypeImport": true,
+        "ignorePrivate": true
+      }],
     },
   },
-];
+  // Possible refactor to use unicorn (eslint-plugin-unicorn is an ESM-only module)
+  // without having to import() dynamically inside an async function. This causes
+  // node to display this warning:
+  // (node:11996) ExperimentalWarning: Importing JSON modules is an experimental feature and might change at any time
+  // This is cause by: const unicorn = await import("eslint-plugin-unicorn")
```

Comment:
```suggestion
  // This is caused by: const unicorn = await import("eslint-plugin-unicorn")
```

Do we want to make an issue to look into this more?

Reply:
I notices that on Mac OS the warning is not displayed. Anyway, we have to live with this untill we move away form  CommonJS